File: app/src/main/java/com/whoguri/learnmvp/javaBase/BaseFragment.java
package com.whoguri.learnmvp.javaBase;

// whoGuri 3/11/18

import android.content.Context;
import android.support.v4.app.Fragment;

public class BaseFragment extends Fragment {
    public BaseActivityListener activityListener;
    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        activityListener= (BaseActivityListener) context;
    }
}

File: app/src/main/java/com/whoguri/learnmvp/kotlinBase/BaseFragment.kt
package com.whoguri.learnmvp.kotlinBase

// whoGuri 3/11/18

import android.content.Context
import android.support.v4.app.Fragment

open class BaseFragment : Fragment() {
    lateinit var activityListener: BaseActivityListener
    override fun onAttach(context: Context?) {
        super.onAttach(context)
        activityListener = (context as BaseActivityListener?)!!
    }
}
